clang/runtime/dpct-rt/include/dpct/bindless_images.hpp

```diff
@@ -352,8 +765,53 @@ dpct_memcpy(pitched_data src, const sycl::id<3> &src_id,
                            dest->get_handle(), dest_offset, dest->get_desc(),
                            copy_extend);
 }
+
+template <typename T>
+inline bool check_duplicate_entries(int count, T **entries) {
+  std::set<T *> uniqueEntries;
+
+  for (size_t i = 0; i < count; ++i) {
+    auto result = uniqueEntries.insert(entries[i]);
+    if (!result.second) {
+      return true;
+    }
+  }
+
+  return false;
+}
 } // namespace detail
 
+/// Map the resource memories to mem handles
+/// \param [in] count The count of resources to map.
+/// \param [in] handles The external mem wrappers used to map the resources.
+/// \param [in] q The queue used to map the resource with.
+inline void map_resources(int count, external_mem_wrapper **handles,
+                          queue_ptr q_ptr = &get_default_queue()) {
+  if (detail::check_duplicate_entries(count, handles)) {
+    throw std::runtime_error(
```

Comment:
how about do like this,  only map one time for duplicated, and return same handle for duplicated items.  then no warnning needed.

Reply:
Written this logic to emulate CUDA's behavior

Ref: [here](https://docs.nvidia.com/cuda/cuda-runtime-api/group__CUDART__INTEROP.html#group__CUDART__INTEROP_1gad8fbe74d02adefb8e7efb4971ee6322:~:text=in%20stream%20begins.-,If%20resources%20contains%20any%20duplicate%20entries%20then%20cudaErrorInvalidResourceHandle%20is%20returned.,-If%20any%20of)